subprojects/provider/src/main/kotlin/org/gradle/kotlin/dsl/execution/ResidualProgramCompiler.kt

```diff
@@ -213,7 +218,7 @@ class ResidualProgramCompiler(
 
     fun emitStage2ProgramFor(scriptFile: File, originalPath: String) {
 
-        val precompiledScriptClass = compileScript(scriptFile, originalPath, stage2ScriptDefinition)
+        val precompiledScriptClass = compileScript(scriptFile, originalPath, stage2ScriptDefinition, "BODY")
```

Comment:
@bamboo can we get something put around these constants to make it clearer that they should remain stable as much as possible? If the do change, we'll just need to be made aware.

We will know from test breakages, but it would be good for later readers to understand that these are consequential.

Reply:
Makes sense, @ldaley. Done in c42c710.